python/iceberg/api/types/type_util.py

```diff
@@ -121,6 +122,8 @@ def visit(arg, visitor): # noqa: ignore=C901
                 result = None
                 try:
                     result = visit(field.type, visitor)
+                except ValidationException as ve:
```

Comment:
This looks strange to me. Why do we catch and raise `ValidationException` but swallow `RuntimeError`. Catching a `RuntimeError` smells like a generic Java conversion as `RuntimeError` isn't a catch-all like `RuntimeException`. @TGooch44 do you know the history here?

Reply:
I'm going to have to step through this because I don't really remember why this is like this.

Reply:
It might be better to always not swallow any errors. I can remove `except` blocks and let the caller to handle the raised errors.

Reply:
Ah, I probably know why. There are many code paths with `NotImplementedError`. If not ignoring errors, multiple tests fail and the build won't pass.
Also, looks like the visitor needs some rework to use `before` and `after` methods (similar to Java). I feel we probably can leave it as it is and resolve that after the rework is done and missing functions are implemented.

Reply:
cool, can we make a comment in the code? Or see if the tests pass if we remove `RuntimeError`. That exception type is a bit worrying for me

Reply:
@rymurr  that is a good point. I updated it to only pass `NotImplementedError` errors and put a comment there.